CoreFeatures/Database-postgresql/PostgresSchema/src/main/java/info/smart_tools/smartactors/database_postgresql/postgres_schema/indexes/IndexCreators.java

```diff
@@ -73,19 +75,25 @@ private static void writeCreateIndex(final String indexType, final Writer body,
         }
 
         List<FieldPath> fieldPaths = new ArrayList<>();
-        if (indexFields instanceof String) {
-            fieldPaths.add(PostgresFieldPath.fromString((String) indexFields));
-        } else if (indexFields instanceof List) {
-            for (Object fieldName : (List) indexFields) {
-                fieldPaths.add(PostgresFieldPath.fromString((String) fieldName));
-            }
-        } else if (indexFields == null) {
-            // ignoring absence of this index type definition
+        if (indexFields == null) {
             return;
-        } else {
-            throw new QueryBuildException("Unknown index definition for " + indexType + ": " + indexFields);
+            // ignoring absence of this index type definition
+        }
+        boolean pathFound = checkAndAddSingleFieldPath(fieldPaths, indexFields);
+        if (!pathFound) {
+            try {
+                for (Object fieldName : (List) indexFields) {
+                    pathFound = checkAndAddSingleFieldPath(fieldPaths, fieldName);
+                    if (!pathFound) {
+                        throw new QueryBuildException("Bad options for creating index task: field data need to be a string, an IObject or a list");
+                    }
+                }
+            } catch (ClassCastException e) {
```

Comment:
Where could it be thrown? If it could be thrown in checkAndAddSingleFieldPath, then why 1st call above is not covered?